### PR TITLE
added stdin

### LINKED ISSUE #252: Implement org.eolang.io.stdin
The `org.eolang.io.stdin` should datarize to the input the user typed into the CLI.

### PR DESCRIPTION
**Pull request by nlchar**
Fixes #252

### CODE REVIEW

**Comment by nlchar** [comment]
@yegor256 should I write tests for this task? And how do I do it? (one of ideas [here](https://pastebin.com/JFGQuXC9))

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `REV` is banned at [#253](https://github.com/cqfn/eo/issues/253); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by yegor256** [comment]
@nlchar of course. See how `stdout` is tested and do something similar.

**Comment by nlchar** [comment]
@yegor256 added a test case for `stdin`. The test mocks the `System.in` stream and checks that `stdin` gathers input string correctly. 
Please, have a look.

**Comment by 0crat** [comment]
You @nlchar[/z](https://www.0crat.com/u/nlchar) need to have one of these roles in order to do "Request order start": `ARC`, `PO`; I'm sorry to say this, but at the moment you've got no roles in this project

File: eo-runtime/src/main/eo/org/eolang/io/stdin.eo [hunk 1]
```diff
@@ -0,0 +1,4 @@
++package org.eolang.io
++rt jvm org.eolang:eo-runtime:0.1.18
+
+[] > stdin /string
```
Comment (yegor256) [inline eo-runtime/src/main/eo/org/eolang/io/stdin.eo:4]: @nlchar I think it's a wrong design. I suggest this one:

```
[] > stdin
  [] > nextLine /string
  [] > @ /string
```

I can read a single line with `stdin.nextLine` or I can take all of it with just `stdin`.

Make sense?

**Comment by yegor256** [comment]
@nlchar see my comment above

**Comment by nlchar** [comment]
@yegor256 should `stdin.@` datarize to an array of lines or just big raw string with `'\n'`'s?

**Comment by yegor256** [comment]
@nlchar just big string with the entire content inside

File: eo-runtime/src/main/eo/org/eolang/io/stdin.eo (hunk 1 above)
Comment (nlchar) [reply eo-runtime/src/main/eo/org/eolang/io/stdin.eo]: @yegor256 done

**Comment by nlchar** [comment]
@yegor256 please have a look.

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `REV` is banned at [#253](https://github.com/cqfn/eo/issues/253); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `REV` is banned at [#253](https://github.com/cqfn/eo/issues/253); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by nlchar** [comment]
Updated the PR to the latest state of the codebase. 

@yegor256 please, have a look.

File: eo-runtime/src/main/java/org/eolang/io/EOstdin$EOnextLine.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * @since 0.1
+ */
+public class EOstdin$EOnextLine extends PhDefault {
+
+    public EOstdin$EOnextLine(final Phi parent) {
+        super(parent);
+        this.add("φ", new AtBound(new AtLambda(this, self -> {
+            try (Scanner sc = new Scanner(System.in)) {
+                if (sc.hasNextLine()) {
+                    return new Data.ToPhi(sc.nextLine());
```
Comment (yegor256) [inline eo-runtime/src/main/java/org/eolang/io/EOstdin$EOnextLine.java:47]: @nlchar see https://www.yegor256.com/2015/01/21/if-then-throw-else.html

File: eo-runtime/src/test/java/org/eolang/io/EOstdinTest.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        MatcherAssert.assertThat(
+            gatheredInput,
+            Matchers.equalTo(input)
+        );
+    }
+
+    @Test
+    public void stdinEmptyTest() throws Exception {
+        String input = "";
+        mockSystemIn(input);
```
Comment (yegor256) [inline eo-runtime/src/test/java/org/eolang/io/EOstdinTest.java:126]: @nlchar this is pretty dirty. You set it, but never reset back. What if this conflicts with other tests?

**Comment by yegor256** [comment]
@nlchar two comments above

File: eo-runtime/src/test/java/org/eolang/io/EOstdinTest.java (hunk 3 above)
Comment (nlchar) [reply eo-runtime/src/test/java/org/eolang/io/EOstdinTest.java:126]: @yegor256 Yegor, I reset System.in in the `@AfterAll rollbackChangesToStdin method of this test class`

File: eo-runtime/src/main/java/org/eolang/io/EOstdin$EOnextLine.java (hunk 2 above)
Comment (nlchar) [reply eo-runtime/src/main/java/org/eolang/io/EOstdin$EOnextLine.java]: @yegor256 totally agree. I did it like this before and forgot this way of handling all bad paths first and leaving the good path after it. 
Did I fix it correctly?

**Comment by nlchar** [comment]
> @nlchar two comments above

@yegor256 thanks for the review. I've tried to fix & answer the issues you noticed.

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `REV` is banned at [#253](https://github.com/cqfn/eo/issues/253); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by nlchar** [comment]
@yegor256 please have a look

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `REV` is banned at [#253](https://github.com/cqfn/eo/issues/253); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `REV` is banned at [#253](https://github.com/cqfn/eo/issues/253); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by nlchar** [comment]
Moving to https://github.com/HSE-Eolang/eo

**Comment by 0crat** [comment]
Job `gh:cqfn/eo#253` is not assigned, can't get performer

**Comment by 0crat** [comment]
Pull request [#253](https://github.com/cqfn/eo/issues/253) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Pull request closed without merging**